apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerDate.cs

```diff
@@ -0,0 +1,408 @@
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// <paramref name="text"/> must be a representation of a date value which can be represented by <see cref="SpannerDate"/>
+        /// </para>
+        /// </remarks>
+        /// <param name="text">The text to parse. Must not be null.</param>
+        /// <returns>The parsed value.</returns>
+        /// <exception cref="FormatException">The value could not be parsed as a <see cref="SpannerDate"/>.</exception>
+        public static SpannerDate Parse(string text)
+        {
+            return Parse(text, CultureInfo.CurrentCulture);
+        }
+
+        /// <summary>
+        /// Parses a textual representation as a <see cref="SpannerDate"/>.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// <paramref name="text"/> must be a representation of a date value which can be represented by <see cref="SpannerDate"/>
+        /// </para>
+        /// </remarks>
+        /// <param name="text">The text to parse. Must not be null.</param>
+        /// <param name="formatProvider">The format provider.</param>
+        /// <returns>The parsed value.</returns>
+        /// <exception cref="FormatException">The value could not be parsed as a <see cref="SpannerDate"/>.</exception>
+        public static SpannerDate Parse(string text, IFormatProvider formatProvider)
+        {
+            string message = TryParseImpl(text, formatProvider, out var value);
+            return message == null ? value : throw new FormatException(message);
+        }
+
+        /// <summary>
+        /// Attempts to parse a textual representation of a <see cref="SpannerDate"/> value.
+        /// </summary>
+        /// <remarks>
+        /// See <see cref="Parse(string, IFormatProvider)"/> for format details. This method will return <c>true</c> if and only if
+        /// <see cref="Parse(string, IFormatProvider)"/> would return without an exception.
+        /// </remarks>
+        /// <param name="text">The text to parse. Must not be null.</param>
+        /// <param name="formatProvider">The format provider. </param>
+        /// <param name="value">The parsed value, or 0 on failure.</param>
+        /// <returns><c>true</c> if <paramref name="text"/> was parsed successfully; <c>false</c> otherwise.</returns>
+        public static bool TryParse(string text, IFormatProvider formatProvider, out SpannerDate value)
+        {
+            string message = TryParseImpl(text, formatProvider, out value);
+            return message == null;
+        }
+
+        /// <summary>
+        /// Returns the error message for a FormatException if parsing fails, or null for success.
+        /// </summary>
+        private static string TryParseImpl(string text, IFormatProvider formatProvider, out SpannerDate value)
+        {
+            // TODO: Do we implement the 3 overloads of Parse just like DateTime does? One taking just a string input,
```

Comment:
I don't think we should have a DateTimeStyles, but I'm somewhat concerned that we can parse whole date/time values rather than just dates. Let's discuss.

(We obviously don't want to implement full date/time parsing ourselves, but we could potentially restrict this to "only ISO-8601" completely. If folks want to parse anything else, they can go via DateTime...)